### ISSUE TITLE
Game::PlaceFood has a bug

### ISSUE THREAD

**Issue by Horki** [udacity/CppND-Capstone-Snake-Game#3]
https://github.com/udacity/CppND-Capstone-Snake-Game/blob/master/src/game.cpp#L60-#L64

If the randomize for "food" is currently on snake, there is no more food generated.
Qucikfix I've build.

```cpp
void Game::PlaceFood() {
  int x, y;
  while (true) {
    x = random_w(engine);
    y = random_h(engine);
    // Check that the location is not occupied by a snake item before placing
    // food.
    if (!snake.SnakeCell(x, y)) {
      food.x = x;
      food.y = y;
      return;
    } else {
      continue;
    }
  }
}
```

**Comment by hauptiness** [comment]
The problem is not in the `Game::PlaceFood()` function, the problem is in the `std::uniform_int_distribution<int> random_w` and `std::uniform_int_distribution<int> random_h`.
They are initialized to generate numbers from `0` to `grid_width` and to `grid_height` respectively. When the random number happens to be exactly `grid_width` or `grid_height` it is outside the grid since the grid numbering starts at `0`.
 Adding `-1` to both should fix it:
```c++
Game::Game(std::size_t grid_width, std::size_t grid_height)
    : snake(grid_width, grid_height),
      engine(dev()),
      random_w(0, static_cast<int>(grid_width)-1),
      random_h(0, static_cast<int>(grid_height)-1) {
  PlaceFood();
}
```

**Comment by hauptiness** [comment]
This has been fixed by #10 but is not merged yet.

**Comment by SudKul** [comment]
https://github.com/udacity/CppND-Capstone-Snake-Game/pull/10 has now been merged.
